### ISSUE TITLE
seccomp tutorial does not list Docker prerequisite

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#31536]
**This is a Bug Report**

**Problem:**
[Restrict a Container's Syscalls with seccomp](https://kubernetes.io/docs/tutorials/security/seccomp/) requires Docker so that you can run `kind`. The requirement is not explicit; if you had `kind` set up OK but using Podman, then you have met the prerequisites but the page does not work. 

**Proposed Solution:**
List Docker as a prerequisite. Explain that you could use a tool such as `podman` instead, but would need to adapt the instructions accordingly.

When clarifying, also consider that readers might be thinking of Docker as the container runtime that runs inside Kubernetes. In fact, `kind` typically interacts with a CRI-compatible runtime inside the cluster that it deploys for you, and _doesn't_ run Docker inside the cluster - see the banner at the top of https://kind.sigs.k8s.io/

**Page to Update:**
https://kubernetes.io/docs/tutorials/security/seccomp/

**Additional Information:**
/language en

**Comment by mtardy** [comment]
I'm interested in this issue, I recently used this tutorial to experiment with Seccomp in order to write some texts about it in the future security checklist we prepare with sig-security-docs.

I actually used `minikube` and it has the particularity of being compatible with [multiple providers](https://minikube.sigs.k8s.io/docs/drivers/) to run the cluster, like Docker (exactly like `kind`) but also VirtualBox, VMware, KVM, etc... But the approach is a bit different in terms of the user interface, `kind` uses YAML to declare the configuration and `minikube` uses flags.

So migrating to `minikube` for the tutorial might be a more generic approach (you can even [use podman as a provider](https://minikube.sigs.k8s.io/docs/drivers/podman/) but it's experimental, like with kind) but require many changes.

Do we want to keep a container-based provider only, keep `kind` and make it more generic by proposing to use the experimental podman or should we even move to `minikube` for this tutorial?

**Comment by mtardy** [comment]
I also have another comment, when I experimented with the tutorial, it felt a bit off that the [Enable the use of RuntimeDefault as the default seccomp profile for all workloads](https://kubernetes.io/docs/tutorials/security/seccomp/#enable-the-use-of-runtimedefault-as-the-default-seccomp-profile-for-all-workloads) was the first step. For me it should be moved to the end, so it would be structured like that:
1. Setup phase
1. Create a Pod with a seccomp profile for syscall auditing
1. Create Pod with seccomp profile that causes violation 
1. Create Pod with seccomp profile that only allows necessary syscalls 
1. Create Pod that uses the container runtime default seccomp profile
1. Enable the use of RuntimeDefault as the default seccomp profile for all workloads

So the order would follow the gradual experimentation. For now the last section "Enable the use of RuntimeDefault as the default seccomp profile for all workloads" has been added directly after the setup phase.
